src/client/components/trackDuration/trackDuration.js

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from 'react';
+import 'react-rangeslider/lib/index.css';
+import './trackDuration.css';
+import PropTypes from 'prop-types';
+import Slider from 'react-rangeslider';
+import { connect } from 'react-redux';
+import VolumeComponent from '../volumeComponent/volumeComponent';
+import SwitcherSound from '../switcherSound/switcherSound';
+
+class TrackDuration extends Component {
+    state = {
+      duration: null,
+      currentTime: null,
+      playing: false,
+      startPlayTime: null,
+      onToggle: true,
+    }
+
+    static getDerivedStateFromProps(props, state) {
+      const { audioData: { loading, sound, trackName } } = props;
+      if (sound && sound.sourceNode) {
+        if (sound && sound.sourceNode && !state.startPlayTime && sound.playing) {
```

Comment:
No need to duplicate sound && sound.sourceNode in nested if statement, because if it won’t exist it won’t reach this code

Reply:
done